client/src/api.rs

```diff
@@ -176,21 +177,26 @@ pub struct DiskAttachment {
     pub state: DiskAttachmentState,
 }
 
+/// A stable index which is translated by Propolis
+/// into a PCI BDF, visible to the guest.
+#[derive(Copy, Clone, Deserialize, Serialize, JsonSchema)]
+pub struct PciSlot(pub u8);
```

Comment:
Perhaps even leave PCI out of the name?  It is kind of implied by the device types we're using. :shrug: